packages/apollo-codegen-typescript/src/codeGeneration.ts

```diff
@@ -246,6 +335,68 @@ export class TypescriptAPIGenerator extends TypescriptGenerator {
     this.scopeStackPop();
   }
 
+  public getGlobalTypesUsedForOperation(doc: Operation | Fragment, context: CompilerContext) {
+    let docTypesUsed: GraphQLType[] = [];
+
+    if (doc.hasOwnProperty('operationName')) {
```

Comment:
Could this be replaced by an `instanceof` or `kind` check? Then it wouldn't be necessary to explicitly cast the value later.